bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/handler/SensiboSkyHandler.java

```diff
@@ -0,0 +1,496 @@
+            if (sensiboSky.isAlive()) {
+                updateStatus(ThingStatus.ONLINE); // In case it has been offline
+                switch (channelUID.getId()) {
+                    case CHANNEL_CURRENT_HUMIDITY:
+                        handleCurrentHumidityCommand(channelUID, command, sensiboSky);
+                        break;
+                    case CHANNEL_CURRENT_TEMPERATURE:
+                        handleCurrentTemperatureCommand(channelUID, command, sensiboSky);
+                        break;
+                    case CHANNEL_MASTER_SWITCH:
+                        handleMasterSwitchCommand(channelUID, command, sensiboSky);
+                        break;
+                    case CHANNEL_TARGET_TEMPERATURE:
+                        handleTargetTemperatureCommand(channelUID, command, sensiboSky);
+                        break;
+                    case CHANNEL_MODE:
+                        handleModeCommand(channelUID, command, sensiboSky);
+                        break;
+                    case CHANNEL_SWING_MODE:
+                        handleSwingCommand(channelUID, command, sensiboSky);
+                        break;
+                    case CHANNEL_FAN_LEVEL:
+                        handleFanLevelCommand(channelUID, command, sensiboSky);
+                        break;
+                    case CHANNEL_TIMER:
+                        handleTimerCommand(channelUID, command, sensiboSky);
+                        break;
+                    default:
+                        logger.debug("Received command on unknown channel {}, ignoring", channelUID.getId());
+
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Unreachable by Sensibo servers");
+            }
+        });
+    }
+
+    private void handleTimerCommand(ChannelUID channelUID, Command command, SensiboSky sensiboSky) {
+        if (command instanceof RefreshType) {
+            if (sensiboSky.getTimer().isPresent() && sensiboSky.getTimer().get().secondsRemaining > 0) {
+                updateState(channelUID, new DecimalType(sensiboSky.getTimer().get().secondsRemaining));
+            } else {
+                updateState(channelUID, UnDefType.UNDEF);
+            }
+        } else if (command instanceof DecimalType) {
+            final DecimalType newValue = (DecimalType) command;
+            updateTimer(newValue.intValue());
+        } else {
+            updateTimer(null);
+        }
+    }
+
+    private void handleFanLevelCommand(ChannelUID channelUID, Command command, SensiboSky sensiboSky) {
+        if (command instanceof RefreshType) {
+            if (sensiboSky.getAcState().isPresent() && sensiboSky.getAcState().get().getFanLevel() != null) {
+                updateState(channelUID, new StringType(sensiboSky.getAcState().get().getFanLevel()));
+            } else {
+                updateState(channelUID, UnDefType.UNDEF);
+            }
```

Comment:
It is tempting, but your suggestion does not compile!

Reply:
Well shoot, what part of it did it not like? It *should* have worked as is, but I'm sure if you wanted to it would still work if you figured out which lambda I screwed up. My guess is it didn't like the `AcState::getFanLevel`...
But once again this suggestion can be ignored...but I'm curious why it didn't work...